src/modules/router/specs/router.transactions.service.spec.ts

```diff
@@ -76,23 +79,57 @@ describe('RouterService', () => {
         );
 
         const transaction = await service.createPair(
-            Address.Zero().bech32(),
+            senderAddress,
             'TOK3-3333',
             'TOK4-123456',
         );
         expect(transaction).toEqual({
             nonce: 0,
             value: '0',
-            receiver: Address.fromHex(
+            receiver: Address.newFromHex(
                 '0000000000000000000000000000000000000000000000000000000000000011',
-            ).bech32(),
-            sender: '',
+            ).toBech32(),
+            sender: senderAddress,
             receiverUsername: undefined,
             senderUsername: undefined,
             gasPrice: 1000000000,
             gasLimit: gasConfig.router.createPair,
             data: encodeTransactionData(
-                'createPair@TOK3-3333@TOK4-123456@erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu',
+                `createPair@TOK3-3333@TOK4-123456@${senderAddress}`,
+            ),
+            chainID: mxConfig.chainID,
+            version: 2,
+            options: undefined,
+            signature: undefined,
+            guardian: undefined,
+            guardianSignature: undefined,
+        });
+    });
+
+    it('should get upgrade pair transaction', async () => {
+        const service = module.get<RouterTransactionService>(
+            RouterTransactionService,
+        );
+
+        const transaction = await service.upgradePair(
+            senderAddress,
+            'WEGLD-123456',
+            'USDC-123456',
+            [10000, 10000],
```

Comment:
use [300, 100] to have values close to real scenario

Reply:
done